commit 7253bbeed7f6d00bd796019d79dc1fe0a805fa8e
Author: David Sommerseth <davids@redhat.com>
Date:   Wed May 26 15:39:19 2010 +0200

    Fixed an issue causing SEGV on some hardware when dmi_processor_id() is called
    
    The dmi_processor_id() function did not check the char *version pointer if it
    was NULL before doing strcmp().  On some hardware, *version will be NULL.

diff --git a/src/dmidecode.c b/src/dmidecode.c
index b89c163..4a2e445 100644
--- a/src/dmidecode.c
+++ b/src/dmidecode.c
@@ -1019,7 +1019,6 @@ xmlNode *dmi_processor_id(xmlNode *node, u8 type, const u8 * p, const char *vers
          ** CPUID instruction or another form of identification.
          */
 
-        //. TODO: PyString_FromFormat does not support %x (yet?)...
         dmixml_AddTextChild(data_n, "ID",
                             "%02x %02x %02x %02x %02x %02x %02x %02x",
                             p[0], p[1], p[2], p[3], p[4], p[5], p[6], p[7]);
@@ -1075,7 +1074,7 @@ xmlNode *dmi_processor_id(xmlNode *node, u8 type, const u8 * p, const char *vers
 
                 sig = 2;
 
-        } else if(type == 0x01 || type == 0x02) {
+        } else if(version && (type == 0x01 || type == 0x02)) {
                 /*
                  ** Some X86-class CPU have family "Other" or "Unknown". In this case,
                  ** we use the version string to determine if they are known to
